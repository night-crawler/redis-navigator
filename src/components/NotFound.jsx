import PropTypes from 'prop-types';
import React from 'react';
import { Header, Icon, Segment } from 'semantic-ui-react';


NotFound.propTypes = {
    message: PropTypes.string
};
export default function NotFound(props) {
    const { message = 'Not found' } = props;

    return (
        <Segment>
            <Header as='h1' textAlign='center'>
                <Icon.Group size='large'>
                    <Icon loading={ true } size='big' name='target' color='red' />
                    <Icon corner={ true } size='big' name='detective' />

                    <Icon.Group size='big'>
                        <Icon loading={ true } size='big' name='target' color='red' />
                        <Icon corner={ true } loading={ true } size='tiny' name='find' color='green' />

                        <Icon.Group size='small'>
                            <Icon loading={ true } size='big' name='target' color='red' />
                            <Icon name='detective' />
                        </Icon.Group>

                    </Icon.Group>
                </Icon.Group>
                <Header.Content>
                    404. { message }
                </Header.Content>

            </Header>


        </Segment>
    );
}
