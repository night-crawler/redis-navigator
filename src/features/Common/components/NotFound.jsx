import PropTypes from 'prop-types';
import React from 'react';
import { Header, Icon, Segment } from 'semantic-ui-react';
import { FormattedMessage as Tr } from 'react-intl';

import messages from '../messages';


NotFound.propTypes = {
  message: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
};
export default function NotFound(props) {
  const { message = <Tr { ...messages.notFound } /> } = props;

  return (
    <Segment className='NotFound'>
      <Header as='h1' textAlign='center'>
        <Icon.Group size='large'>
          <Icon loading={ true } size='big' name='target' color='red' />
          <Icon corner={ true } size='big' name='detective' />

          <Icon.Group size='big'>
            <Icon loading={ true } size='big' name='target' color='red' />
            <Icon corner={ true } loading={ true } size='tiny' name='find' color='green' />

            <Icon.Group size='small'>
              <Icon loading={ true } size='big' name='target' color='red' />
              <Icon name='detective' />
            </Icon.Group>

          </Icon.Group>
        </Icon.Group>
        <Header.Content>
                    404. { message }
        </Header.Content>
      </Header>
    </Segment>
  );
}
