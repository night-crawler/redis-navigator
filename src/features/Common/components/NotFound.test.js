/* eslint-disable react/react-in-jsx-scope,no-undef */
import { NotFound } from './NotFound';

describe('NotFound', () => {
  it('should render NotFound', () => {
    const wrapper = shallow(
      <NotFound message='lol' />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
