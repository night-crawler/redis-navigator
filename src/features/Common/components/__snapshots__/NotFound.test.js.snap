// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render NotFound 1`] = `
<Segment
  className="NotFound"
>
  <Header
    as="h1"
    textAlign="center"
  >
    <IconGroup
      as="i"
      size="large"
    >
      <Icon
        as="i"
        color="red"
        loading={true}
        name="target"
        size="big"
      />
      <Icon
        as="i"
        corner={true}
        name="detective"
        size="big"
      />
      <IconGroup
        as="i"
        size="big"
      >
        <Icon
          as="i"
          color="red"
          loading={true}
          name="target"
          size="big"
        />
        <Icon
          as="i"
          color="green"
          corner={true}
          loading={true}
          name="find"
          size="tiny"
        />
        <IconGroup
          as="i"
          size="small"
        >
          <Icon
            as="i"
            color="red"
            loading={true}
            name="target"
            size="big"
          />
          <Icon
            as="i"
            name="detective"
          />
        </IconGroup>
      </IconGroup>
    </IconGroup>
    <HeaderContent>
      404. 
      lol
    </HeaderContent>
  </Header>
</Segment>
`;
